### DISCUSSION TITLE
Tailwind print-only styles not working

### DISCUSSION THREAD

**Discussion by Synchro** [tailwindlabs/tailwindcss#12887] [Help]
I'm using a very vanilla tailwind setup, with a few colours as the only customisation. I have a section in a document that I only want to appear when rendering for print, that's styled like this, in accordance with [tailwind docs](https://tailwindcss.com/docs/hover-focus-and-other-states#print-styles):

```html
<p class="hidden print:block">hello</p>
```
This isn't working – it's always hidden. Other print-only styles don't seem to work either, for example this doesn't print in red:

```html
<p class="print:text-red-600">hello</p>
```

But this *is* only hidden when printing:

```html
<p class="print:hidden">hello</p>
```

I am generating PDFs using [Spatie's BrowserShot](https://spatie.be/docs/browsershot/v4/introduction), and using `emulateMedia('print')`, but I don't think that has anything to do with it as it also fails to heed these styles when I use a browser's built-in print function.

Any idea why these print styles are not working?

**Comment by cubedhuang** [comment]
For me, it was fixed after making the styles `!important` using the `!` modifier:

```html
<p class="hidden print:!block">hello</p>
```

```html
<p class="print:!text-red-600">hello</p>
```

Although I'm not sure why there's an inconsistency -- sometimes it'll work without it and sometimes it won't.

**Comment by ecstrema** [comment]
For those looking to a tailwind4 answer, add this to your style.css:

```css
@custom-variant print (@media print);
```

**Reply by wongjn** [reply]
You shouldn't have to do this; [`print` is built in](https://tailwindcss.com/docs/hover-focus-and-other-states#print).

**Reply by ecstrema** [reply]
Oh wow, I missed it.

**Reply by ecstrema** [reply]
Hence why it worked though:)
